### PR TITLE
Adding resource request proxy

### PR DESCRIPTION
**Pull request by gregmccoy**
## Summary

Updating proxy for new resource request APIs in project service. Uses the current format of the resource requests in dev, will be updated later to reflect the changes in PILOT-1213

## JIRA Issues

PILOT-1116

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

## Test Directions

Call resource request APIs

### CODE REVIEW

File: models/resource_request.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from pydantic import BaseModel
+
+
+class CreateResourceRequest(BaseModel):
+    project_id: str
+    request_for: str
```
Comment (hugoantunes) [inline models/resource_request.py:6]: those fields are also required to create resource requests:
-     user_id: str
-     email: EmailStr
-     username: str

File: tests/api/api_resource_request/test_resource_request.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+from uuid import uuid4
+import pytest
+
+
+RESOURCE_REQUEST = {
+    "project_id": str(uuid4()),
+    "requested_by_user_id": str(uuid4()),
+    "requested_for": "SuperSet",
+    "completed_at": "2022-07-05T14:27:14.834812+00:00",
+    "id": str(uuid4())
```
Comment (hugoantunes) [inline tests/api/api_resource_request/test_resource_request.py:26]: username and email are also required.

**Review by hugoantunes — CHANGES_REQUESTED** [review]

File: models/resource_request.py (hunk 1 above)
Comment (gregmccoy) [reply models/resource_request.py:6]: I grab all those fields from user token so they aren't actually required here

File: api/api_resource_request/resource_request.py [hunk 3]
```diff
@@ -191,120 +236,76 @@ class ResourceRequests:
...
-        "username": username,
-    }
-    response = requests.get(ConfigClass.AUTH_SERVICE + "admin/users/realm-roles", params=payload)
-    if not response.json():
-        raise APIException(status_code=EAPIResponseCode.not_found.value, error_msg="User not found in keycloak")
-    return response.json()["result"]
+        payload = {
+            "project_id": data.project_id,
+            "requested_by_user_id": self.current_identity["user_id"],
+            "requested_for": data.request_for,
```
[the remark this answers is not in the document]
Comment (gregmccoy) [reply api/api_resource_request/resource_request.py:264]: Updated

**Review by hugoantunes — APPROVED** [review]

**Pull request merged**